### ISSUE TITLE
New-idservice gets new ID but doesn't save it.

### ISSUE THREAD

**Issue by gjost** [densho/ddr-local#211]
Redirected from https://github.com/densho/ddr-idservice/issues/2

Archivist created entity ddr-densho-325-210 using ddr-local, entered metadata and added master/mezz binaries.
She then clicked "create new object" but the "new" object ID was ddr-densho-325-210 again.
All the fields were blank, no master/mezz files present in the web ui. Master/mezz files were present in the back end.
No error messages. Bug has been replicated following the above steps.

Looks like ID service is issuing new IDs but not adding them to its backend store.
That is, the service will continue to issue the same ID over and over again to the ddr-editor webui.
If ID is added manually through the webadmin, it will issue the next ID properly.

**Comment by gjost** [comment]
In `ddrlocal.views.entities.new_idservice` the `register` flag should have been set to `True` but is not currently being set at all. This causes the client to hit `ddr-idservice`'s `next_object` view with `GET` instead of `POST`; the next ID object is instantiated but never saved (the behavior described above).

This flag was intended to be a way for clients to see what the next ID would be without affecting the database.

I've added that setting and run it in my local and `idservice` now records the requested new IDs. I'm going to investigate when/how/why this was changed and should have a fix shortly.
